presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -330,16 +375,20 @@ public JdbcOutputTableHandle beginCreateTable(ConnectorTableMetadata tableMetada
     }
 
     @Override
-    public void commitCreateTable(JdbcOutputTableHandle handle, Collection<Slice> fragments)
+    public void finishInsertTable(JdbcInsertTableHandle handle, Collection<Slice> fragments)
     {
-        StringBuilder sql = new StringBuilder()
-                .append("ALTER TABLE ")
-                .append(quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTemporaryTableName()))
-                .append(" RENAME TO ")
-                .append(quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTableName()));
+        StringBuilder insertSql = new StringBuilder();
+        StringBuilder cleanTempTableSql = new StringBuilder();
+        insertSql.append("INSERT INTO ")
+                .append(quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTableName()))
+                .append(" SELECT * FROM ")
+                .append(quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTemporaryTableName()));
+        cleanTempTableSql.append("DROP TABLE ")
+                .append(quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTemporaryTableName()));
 
         try (Connection connection = getConnection(handle)) {
-            execute(connection, sql.toString());
+            execute(connection, insertSql.toString());
```

Comment:
The INSERT/DROP TABLE should be in a transaction, so that if Presto reports that the insert failed because DROP TABLE  failed, the data is not present in the target table.